subcommand/injector/command.go

```diff
@@ -368,20 +368,25 @@ func (c *Command) certWatcher(ctx context.Context, ch <-chan cert.Bundle, client
 	}
 }
 
-func (c *Command) updateWebhook(ctx context.Context, clientset *kubernetes.Clientset, bundle cert.Bundle, webhooksCache cache.Store, leaderElector leader.Elector, log hclog.Logger) error {
+func (c *Command) fetchWebhook(ctx context.Context, clientset *kubernetes.Clientset) (*adminv1.MutatingWebhookConfiguration, error) {
+	return clientset.AdmissionregistrationV1().MutatingWebhookConfigurations().Get(ctx, c.flagAutoName, metav1.GetOptions{})
+}
+
+func (c *Command) getWebhookCached(webhooksCache cache.Store) (*adminv1.MutatingWebhookConfiguration, error) {
 	item, exists, err := webhooksCache.GetByKey(c.flagAutoName)
 	if !exists {
-		log.Warn("Could not find webhook config in cache. Trying again...", "item", item)
-		return fmt.Errorf("could not find webhook config in cache")
+		return nil, fmt.Errorf("webhook config not cached")
 	} else if err != nil {
-		log.Warn(fmt.Sprintf("Could not find webhook config in cache: %s. Trying again...", err))
-		return err
+		return nil, err
 	}
 	config, ok := item.(*adminv1.MutatingWebhookConfiguration)
 	if !ok {
-		log.Error("Got unknown object from cache; expected &MutatingWebhookConfiguration", "item", item)
+		return nil, fmt.Errorf("got unknown object from cache; expected &MutatingWebhookConfiguration but got %T", item)
 	}
```

Comment:
As I was poking around in this, I noticed that if we were to use the Lister() interface from the MutatingWebhookConfigurationInformer instead of Informer().GetStore(), that `Lister().Get` would give back the typed adminv1.MutatingWebhookConfiguration object without needing to assert it here. (Just a thought.)

Reply:
Oh cool. I'll switch to that.

Reply:
Oh, but then we wouldn't be reading from the cache, which we know is up-to-date, but then fetching it again in the `Lister()`, which isn't ideal. (It could be out of sync even.) I think it's probably best to stick with the cache.

Reply:
Oh? I thought the Lister was giving back essentially an indexed view on the cache.

https://github.com/kubernetes/client-go/blob/60a0346672170c8e9f65795fb33e41527980c583/informers/admissionregistration/v1/mutatingwebhookconfiguration.go#L87

vs

https://github.com/kubernetes/client-go/blob/60a0346672170c8e9f65795fb33e41527980c583/informers/admissionregistration/v1/mutatingwebhookconfiguration.go#L87

Reply:
Ah, you appear correct. Will fix.